src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java

```diff
@@ -44,58 +55,100 @@ public TenantCollection get(UUID consortiumId, Integer offset, Integer limit) {
   }
 
   @Override
+  public String getCentralTenantId() {
+    TenantEntity tenant = tenantRepository.findCentralTenant()
+      .orElseThrow(() -> new ResourceNotFoundException("A central tenant is not found. The central tenant must be created"));
+    return tenant.getId();
+  }
+
+  @Override
+  public TenantEntity getByTenantId(String tenantId) {
+    return tenantRepository.findById(tenantId)
+      .orElse(null);
+  }
+
+  @Override
+  @Transactional
   public Tenant save(UUID consortiumId, Tenant tenantDto) {
-    consortiumService.checkConsortiumExistsOrThrow(consortiumId);
-    checkTenantNotExistsOrThrow(tenantDto.getId());
-    TenantEntity entity = toEntity(consortiumId, tenantDto);
-    TenantEntity tenantEntity = tenantRepository.save(entity);
-    return converter.convert(tenantEntity, Tenant.class);
+    log.debug("save:: Trying to save a tenant by consortiumId '{}', tenant object with id '{}' and isCentral={}", consortiumId, tenantDto.getId(), tenantDto.getIsCentral());
+    FolioExecutionContext currentTenantContext = (FolioExecutionContext) folioExecutionContext.getInstance();
+    String centralTenantId;
+
+    if (tenantDto.getIsCentral()) {
+      checkCentralTenantExistsOrThrow();
+      centralTenantId = tenantDto.getId();
+    } else {
+      centralTenantId = getCentralTenantId();
+    }
+
+    runInFolioContext(createFolioExecutionContextForTenant(tenantDto.getId(), currentTenantContext, folioMetadata),
+      () -> configurationClient.saveConfiguration(createConsortiaConfigurationBody(centralTenantId)));
+    log.info("save:: saved consortia configuration with centralTenantId={} for tenant={} context", centralTenantId, tenantDto.getId());
+    try (var context = new FolioExecutionContextSetter(createFolioExecutionContextForTenant(centralTenantId,
```

Comment:
In order to not switch context twice you can firstly save to DB central tenant and after this switch context to save configuration in the institutional tenant.
Your method annotated with transactional annotation, so if there will be some exceptionon the last step when saving to mod-configuration - changes into central tenant DB also will be reverted

Reply:
Fixed